packages/teams-js/test/public/pages.spec.ts

```diff
@@ -210,42 +335,917 @@ describe('AppSDK-TeamsAPIs', () => {
+        const messageForSettingeRemoveHandler = utils.findMessageByFunc('settings.remove');
+        expect(messageForSettingeSaveHandler).toBeNull();
+        expect(messageForSettingeRemoveHandler).toBeNull();
+      });
+    });
+
+    describe('Testing pages.config.setValidityState function', () => {
+      const allowedContexts = [FrameContexts.settings, FrameContexts.remove];
+
+      it('pages.config.setValidityState should not allow calls before initialization', () => {
+        expect(() => pages.config.setValidityState(true)).toThrowError('The library has not yet been initialized');
+        expect(() => pages.config.setValidityState(false)).toThrowError('The library has not yet been initialized');
+      });
+
+      Object.values(FrameContexts).forEach(context => {
+        if (allowedContexts.some(allowedContexts => allowedContexts === context)) {
+          it(`pages.config.setValidityState should successfully set validity state to true when initialized with ${context} context`, async () => {
+            await utils.initializeWithContext(context);
+            pages.config.setValidityState(true);
+
+            const message = utils.findMessageByFunc('settings.setValidityState');
+            expect(message).not.toBeNull();
+            expect(message.args.length).toBe(1);
+            expect(message.args[0]).toBe(true);
+          });
+
+          it(`pages.config.setValidityState should successfully set validity state to false when initialized with ${context} context`, async () => {
+            await utils.initializeWithContext(context);
+            pages.config.setValidityState(false);
+
+            const message = utils.findMessageByFunc('settings.setValidityState');
+            expect(message).not.toBeNull();
+            expect(message.args.length).toBe(1);
+            expect(message.args[0]).toBe(false);
+          });
+        } else {
+          it(`pages.config.setValidityState does not allow calls from ${context} context`, async () => {
+            await utils.initializeWithContext(context);
+            expect(() => pages.config.setValidityState(true)).toThrowError(
+              `This call is only allowed in following contexts: ${JSON.stringify(
+                allowedContexts,
+              )}. Current context: "${context}".`,
+            );
+          });
+        }
+      });
+    });
+
+    describe('Testing pages.config.setConfig function', () => {
+      const allowedContexts = [FrameContexts.content, FrameContexts.settings, FrameContexts.sidePanel];
+
+      let settingsObj: pages.InstanceConfig = {
+        suggestedDisplayName: 'someSuggestedDisplayName',
+        contentUrl: 'someContentUrl',
+        websiteUrl: 'someWebsiteUrl',
+        entityId: 'someEntityId',
+      };
+
+      it('pages.config.setConfig should not allow calls before initialization', () => {
+        expect(() => pages.config.setConfig({} as pages.InstanceConfig)).rejects.toThrowError(
```

Comment:
It's interesting. I was curious if calling `pages.config.setConfig` with `{} as pages.InstanceConfig` was valid and it turns out it is (which makes me think we should have a test that takes an empty `pages.InstanceConfig`). It's a little surprising because `pages.InstanceConfig` has a required property of `contentUrl`, but you can pass an empty object without that required property. I wonder if that should somehow result in an error?

Reply:
Apparently when passed in empty object we do get empty object in return adding that test case doesn't checks or verifies anything.